### PR TITLE
docs(release): added regression checking and clarifications

### PR DESCRIPTION
**Pull request by jm-clius**
This adds some minor clarifications to the release process instructions for contributors:
- ensuring that CI against js-waku passes against the release candidate
- clarified the manual process by which the tagged Docker image should be published

### CODE REVIEW

File: docs/contributors/release-process.md [hunk 1]
```diff
@@ -56,7 +59,14 @@ git push origin v0.1
 ### After the release
 
 1. Announce the release on Twitter, Discord and other channels.
-2. Deploy the release:
+2. Deploy the release image to [Dockerhub](https://hub.docker.com/layers/statusteam/nim-waku/a5f8b9/images/sha256-88691a8f82bd6a4242fa99053a65b7fc4762b23a2b4e879d0f8b578c798a0e09?context=explore) by triggering [the manual Jenkins deployment job]([https://ci.status.im/job/nim-waku/job/deploy-wakuv2-prod/)](https://ci.infra.status.im/job/nim-waku/job/manual/build).
```
Comment (alrevuelta) [inline docs/contributors/release-process.md:62]: Nitpick but looks like doesn't render well.

**Review by alrevuelta — APPROVED** [review]
lgtm! Great to have this documented.

File: docs/contributors/release-process.md (hunk 1 above)
Comment (jm-clius) [reply docs/contributors/release-process.md:62]: Oh! Thanks. Missed this!

Comment (jm-clius) [reply docs/contributors/release-process.md]: Fixed in [9275e62](https://github.com/waku-org/nwaku/pull/1610/commits/9275e62684e49966ba1280bc8acba32c2ed7727f)

**Pull request merged**